client/src/components/app.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Deals from './Deals.jsx';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      deals: [
```

Comment:
Suggest to add a new field to your DB collection, to track the most recently viewed date/time (per deal). And select the deals to display, using this new field.